src/sdk/pynni/nni/smac_tuner/convert_ss_to_scenario.py

```diff
@@ -41,17 +41,22 @@ def generate_pcs(nni_search_space_content):
     # parameter_name real [min_value, max_value] [default value] log
     # https://automl.github.io/SMAC3/stable/options.html
     '''
+    categorical_dict = {}
     search_space = nni_search_space_content
     with open('param_config_space.pcs', 'w') as pcs_fd:
         if isinstance(search_space, dict):
             for key in search_space.keys():
                 if isinstance(search_space[key], dict):
                     try:
                         if search_space[key]['_type'] == 'choice':
+                            choice_len = len(search_space[key]['_value'])
```

Comment:
Is it possible that user's search space has no 'choice' value?

Reply:
yes, it is. If there is no choice, no need to do convert for categorical in smac